### ISSUE TITLE
Incorrect transformation: OptionalChainExpression

### ISSUE THREAD

**Issue by SukkaW** [lxsmnsyc/forgetti#28]
**Input**

```jsx
function Example(props) {
  const data = props.a?.b.c;

  return <h1>{data}</h1>;
}
```

**Output**

```jsx
import { useRef as _useRef } from "react";
import { $$cache as _$$cache } from "forgetti/runtime";
import { $$equals as _$$equals } from "forgetti/runtime";
import { memo as _memo } from "react";
import { $$memo as _$$memo } from "forgetti/runtime";
const _Memo = _$$memo(_memo, _values => <h1>{_values[0]}</h1>);
function Example(props) {
  let _cache = _$$cache(_useRef, 10),
    _equals = _$$equals(_cache, 0, props),
    _value = _equals ? _cache[0] : _cache[0] = props,
    _value2 = _equals ? _cache[1] : _cache[1] = _value.a,
    _equals2 = _$$equals(_cache, 2, _value2),
    _value3 = _equals2 ? _cache[2] : _cache[2] = _value2,
    _value4 = _equals2 ? _cache[3] : _cache[3] = _value3?.b,
    _equals3 = _$$equals(_cache, 4, _value4),
    _value5 = _equals3 ? _cache[4] : _cache[4] = _value4,
    _value6 = _equals3 ? _cache[5] : _cache[5] = _value5.c,
    _equals4 = _$$equals(_cache, 6, _value6),
    _value7 = _equals4 ? _cache[6] : _cache[6] = _value6,
    _value8 = _equals4 ? _cache[7] : _cache[7] = [_value7],
    _equals5 = _$$equals(_cache, 8, _value8),
    _value9 = _equals5 ? _cache[8] : _cache[8] = _value8;
  return _equals5 ? _cache[9] : _cache[9] = /*@forgetti skip*/<_Memo v={_value9} />;
}
```

----

When `props.a` is `undefined`, `_value2` becomes `undefined`, then `_value3 = _value2` is also `undefined`, `_value4 = _value3?.b` is also undefined, `_value5 = _value4` is also undefined, now `_value6` will error since we can't read `c` on `undefined`(`_value5`).

**Comment by lxsmnsyc** [comment]
Interesting edge case, not sure yet what's the best output for this.

**Comment by lxsmnsyc** [comment]
Should be fixed now. This also fixes inconsistency with optional call expressions (which includes `a?.(b)` and `a.b?.(c)`

**Comment by SukkaW** [comment]
Confirmed the issue has been fixed in 0.6.2. Closing.
